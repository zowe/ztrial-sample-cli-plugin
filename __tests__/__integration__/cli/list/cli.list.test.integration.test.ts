/**
 * This program and the accompanying materials are made available and may be used, at your option, under either:
 * * Eclipse Public License v2.0, available at https://www.eclipse.org/legal/epl-v20.html, OR
 * * Apache License, version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
 *
 * Copyright Contributors to the Zowe Project.
 */

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../__src__/environment/doc/ITestPropertiesSchema";

// Test environment will be populated in the "beforeAll"
let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;

describe("zowe-cli-sample list command", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            installPlugin: true,
            testName: "list_command",
            skipProperties: true
        });
    });
    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });
    it("should display the help", () => {
        const response = runCliScript(__dirname + "/__scripts__/list_help.sh", TEST_ENVIRONMENT);

        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
