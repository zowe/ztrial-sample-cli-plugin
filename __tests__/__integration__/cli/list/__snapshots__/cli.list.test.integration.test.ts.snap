// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zowe-cli-sample list command should display the help 1`] = `
"
 DESCRIPTION
 -----------

   [actions] in Zowe CLI are groups of commands. [actions] are always verbs. For
   example, for command \\"zowe zos-files delete data-set\\", the [action] is \\"delete\\".

   For this action (\\"list\\") we will list files, etc.

 USAGE
 -----

   zowe zowe-cli-sample list <command>

   Where <command> is one of the following:

 COMMANDS
 --------

   directory-contents | dc Lists directory contents
   profile-args | pa       Lists profile args      

 GLOBAL OPTIONS
 --------------

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-examples (boolean)

      Display examples for all the commands in a group

   --help-web | --hw (boolean)

      Display HTML help in browser

"
`;
